Lib/typing.py

```diff
@@ -147,10 +149,13 @@ def _type_check(arg, msg, is_argument=True):
         return arg
     if isinstance(arg, _SpecialForm) or arg in (Generic, Protocol):
         raise TypeError(f"Plain {arg} is not valid as type argument")
-    if isinstance(arg, (type, TypeVar, ForwardRef, types.Union)):
+    if isinstance(arg, (type, TypeVar, ForwardRef, types.Union, ParamSpec)):
         return arg
     if not callable(arg):
         raise TypeError(f"{msg} Got {arg!r:.100}.")
+    if isinstance(arg, _ConcatenateGenericAlias):
+        raise TypeError(f"{arg} is not valid as a type argument "
```

Comment:
```suggestion
        raise TypeError(f"Concatenate[...] is not valid as a type argument "
```
However, maybe it *is* valid? The PEP has examples like
```
def f(x: X[int, Concatenate[int, P_2]]) -> str: ...  # Accepted
```
I would also suggest not to be too strict -- we can leave that to the type checkers. At runtime we don't want to slow things down.